cmd/tmsUpload.go

```diff
@@ -42,7 +42,11 @@ func runTmsUpload(uploadConfig tmsUploadOptions, communicationInstance tms.Commu
 
 func convertUploadOptions(uploadConfig tmsUploadOptions) tms.Options {
 	var config tms.Options
-	config.TmsServiceKey = uploadConfig.TmsServiceKey
+	config.ServiceKey = uploadConfig.ServiceKey
+	if len(uploadConfig.ServiceKey) > 0 {
+		config.ServiceKey = uploadConfig.TmsServiceKey
+		log.Entry().Warn("DEPRECATION WARNING: The tmsServiceKey parameter has been deprecated, please use the serviceKey parameter instead.")
+	}
 	config.CustomDescription = uploadConfig.CustomDescription
 	if config.CustomDescription == "" {
```

Comment:
I'm pretty sure this does not what we want here. The `config.CustomDescription` is always empty, as it was just instantiated a few lines above. Here the `tmsUploadOptions.CustomDescription` should be checked.
I will open a separate PR

Reply:
@artembannikov @gerstneralex https://github.com/SAP/jenkins-library/pull/4679

Reply:
Urgh, I was on a wrong tangent there. Sorry for the confusion, it is set in the line just above :/ Closing the PR